test/unit/collection-view.filter.spec.js

```diff
@@ -0,0 +1,398 @@
+        expect(this.collectionView.children.findByModel(this.model3)).to.exist;
+        expect(this.collectionView.children.findByModel(this.model4)).not.to.exist;
+      });
+
+      it("renders the views in the correct order", function() {
+        expect(this.collectionView.$el).to.contain.$text('13');
+      });
+    });
+
+    describe("when the filter changes between sorts", function() {
+      beforeEach(function() {
+        this.collectionView.render();
+        this.collectionView.filter = this.inverseFilter;
+        this.collection.sort();
+      });
+
+      it("only renders views that pass the filter", function() {
+        expect(this.collectionView.children.findByModel(this.model1)).not.to.exist;
+        expect(this.collectionView.children.findByModel(this.model2)).to.exist;
+        expect(this.collectionView.children.findByModel(this.model3)).not.to.exist;
+        expect(this.collectionView.children.findByModel(this.model4)).to.exist;
+      });
+
+      it("renders the views in the correct order", function() {
+        expect(this.collectionView.$el).to.contain.$text('24');
+      });
+    });
+
+    describe("when the filter changes between sorts and the collection hasn't changed order", function() {
+      beforeEach(function() {
+        this.collection.sort();
+        this.collectionView.render();
+        this.collectionView.filter = this.inverseFilter;
+        this.collection.sort();
+      });
+
+      it("only renders views that pass the filter", function() {
+        expect(this.collectionView.children.findByModel(this.model1)).not.to.exist;
+        expect(this.collectionView.children.findByModel(this.model2)).to.exist;
+        expect(this.collectionView.children.findByModel(this.model3)).not.to.exist;
+        expect(this.collectionView.children.findByModel(this.model4)).to.exist;
+      });
+
+      it("renders the views in the correct order", function() {
+        expect(this.collectionView.$el).to.contain.$text('24');
+      });
+    });
+  });
+
+  describe("combined with a reorder", function() {
+    beforeEach(function() {
+      this.collection.comparator = 'bar';
+      this.model1 = new Backbone.Model({foo: true, bar: 1});
+      this.model2 = new Backbone.Model({foo: false, bar: 2});
+      this.model3 = new Backbone.Model({foo: true, bar: 3});
+      this.model4 = new Backbone.Model({foo: false, bar: 4});
+      this.collection.reset([this.model2, this.model4, this.model3, this.model1]);
+
+      this.ChildView = Marionette.ItemView.extend({
+        template: function(data) {
+          return data.bar;
+        }
+      });
+      this.collectionView = new this.CollectionView({
+        childView: this.ChildView,
+        reorderOnSort: true
+      });
+    });
+
+    describe("when the filter does not change between sorts", function() {
+      beforeEach(function() {
+        this.collectionView.render();
+        this.collection.sort();
```

Comment:
Because `render` is called first I dont actually think the `sort` will trigger a reorder as the order hasn't changed. Maybe try silently adding a model to the start of the collection and then calling `sort`